backend/src/equipment/equipment.service.ts

```diff
@@ -0,0 +1,117 @@
+import { Injectable } from '@nestjs/common';
+import { Equipment } from '@prisma/client';
+import { ScheduleDTO } from '../schedule/dto/schedule.dto';
+import { PrismaService } from '../global/prisma/prisma.service';
+import { EquipmentDTO } from './dto/equipment.dto';
+
+@Injectable()
+export class EquipmentService {
+  constructor(private readonly prisma: PrismaService) {}
+
+  public async addEquipment(data: EquipmentDTO): Promise<EquipmentDTO> {
+    const equipment = await this.prisma.equipment.create({
+      data,
+    });
+    return EquipmentService.convertToDTO(equipment);
+  }
+  public async addScheduletoEquipment(
+    id: string,
+    data: ScheduleDTO
+  ): Promise<EquipmentDTO> {
+    const schedule = await this.prisma.schedule.create({
+      data,
+    });
+    const equipment = await this.prisma.equipment.update({
+      where: {
+        id,
+      },
+      data: {
+        schedule: {
+          connect: {
+            id: schedule.id,
+          },
+        },
+      },
+    });
+    return equipment;
+  }
+  public async getAllEquipments(): Promise<any> {
+    const getAll = await this.prisma.equipment.findMany({
+      include: {
+        schedule: true,
+      },
+    });
+    return getAll;
+  }
+  public async deleteEquipment(id: string): Promise<EquipmentDTO> {
+    const equipment = await this.prisma.equipment.delete({
+      where: {
+        id,
+      },
+    });
+    return EquipmentService.convertToDTO(equipment);
```

Comment:
whenever we get/update/delete an entity by id, we have to wrap it in a try/catch to catch the `NotFoundError` thrown by Prisma. there are some examples in this pr's service class: https://github.com/citu-cpe/pnrn/pull/14/files#diff-71d61c3ecc2e25b05c6cf71aaee6092c3d47f8a03d9b4f05b4a1012b26f9f69e